### PR TITLE
Add pure Java projects support

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

**Comment by alexarchambault** [comment]
(If that can help, I can maybe split the last commit, which is a bit large.)

File: modules/build/src/test/scala/scala/build/tests/BuildTests.scala [hunk 1]
```diff
@@ -897,4 +904,18 @@ abstract class BuildTests(server: Boolean) extends munit.FunSuite {
       expect(scalaCompilerJarNameOpt.contains("scala3-compiler_3-3.1.0.jar"))
     }
   }
+
+  test("Pure Java") {
+    val inputs = TestInputs(
+      os.rel / "Foo.java" ->
+        """
+          |""".stripMargin
```
Comment (Gedochao) [inline modules/build/src/test/scala/scala/build/tests/BuildTests.scala:918]: Do we want more tests for pure Java here?
Maybe smth with actual java code would be worth it?

File: modules/build/src/main/scala/scala/build/compiler/SimpleScalaCompiler.scala [hunk 2]
```diff
@@ -25,70 +25,108 @@ final case class SimpleScalaCompiler(
...
+          mainClassOpt.forall { mainClass =>
+
+            val outputDir =
+              if (isScala2 && scaladoc) project.scaladocDir
+              else project.classesDir
+
+            runScalacLike(project, mainClass, outputDir, logger)
+          }
+
+        case None =>
```
Comment (Gedochao) [inline modules/build/src/main/scala/scala/build/compiler/SimpleScalaCompiler.scala:95]: Thinking if it wouldn't be less confusing if we extracted the pure Java scenario code to a separate class (`SimpleJavaCompiler`?), which could be called from here

**Review by Gedochao — APPROVED** [review]
The other comments are just nitpicks, LGTM

File: modules/build/src/test/scala/scala/build/tests/BuildTests.scala (hunk 1 above)
Comment (alexarchambault) [reply modules/build/src/test/scala/scala/build/tests/BuildTests.scala:918]: Oh yep, that's an oversight.

File: modules/build/src/main/scala/scala/build/compiler/SimpleScalaCompiler.scala (hunk 2 above)
Comment (alexarchambault) [reply modules/build/src/main/scala/scala/build/compiler/SimpleScalaCompiler.scala:95]: Good idea, just did that.

**Pull request merged**